assets/shaders/vat.glsl

```diff
@@ -1,24 +1,22 @@
-#version 130 // required by gl_VertexID
-
-attribute vec2 a_texCoord1; // Texture coordinates
+attribute float a_vat_index;
+attribute vec2 a_texCoord1; // Diffuse Texture coordinates
 
 uniform mat4 u_worldTrans; // World transformation matrix
 uniform mat4 u_projViewTrans; // View transformation matrix
 
 uniform sampler2D u_vertexAnimationTexture; // Texture containing animated vertex positions (float texture)
-uniform float u_textureHeight; // Height of the vertex texture (number of animation frames)
-uniform float u_textureWidth; // Width of the vertex texture (number of vertices)
-uniform float u_frame1; // Index of the first frame in the animation texture
-uniform float u_frame2; // Index of the second frame in the animation texture
+uniform int u_textureHeight; // Height of the vertex texture (number of animation frames)
+uniform int u_textureWidth; // Width of the vertex texture (number of vertices)
+uniform int u_frame1; // Index of the first frame in the animation texture
+uniform int u_frame2; // Index of the second frame in the animation texture
 uniform float u_interpolation; // Interpolation factor between two animation frames (0.0 to 1.0)
 
-varying vec2 v_texCoord;
+varying vec2 v_diffuseUV;
 
 void main() {
-    float index = float(gl_VertexID);
     vec2 texelSize = vec2(1.0 / u_textureWidth, 1.0 / u_textureHeight);
-    vec2 vertexTextureCoord1 = vec2((index + 0.5) * texelSize.x, (u_frame1 + 0.5) * texelSize.y);
-    vec2 vertexTextureCoord2 = vec2((index + 0.5) * texelSize.x, (u_frame2 + 0.5) * texelSize.y);
+    vec2 vertexTextureCoord1 = vec2((a_vat_index + 0.5) * texelSize.x, (u_frame1 + 0.5) * texelSize.y);
+    vec2 vertexTextureCoord2 = vec2((a_vat_index + 0.5) * texelSize.x, (u_frame2 + 0.5) * texelSize.y);
```

Comment:
An article, outlining using "vertex attribute textures" and difficulties you ran into when doing this, related to libGDX, java and opengl might make an interesting read. I didn't find one in the wild right off the bat.

Reply:
added a ticket